### ISSUE TITLE
Is there any range limitations of  'timesteps' in the LSTM input?

### ISSUE THREAD

**Issue by Yingyingzhang15** [fchollet/keras#2057]
Hi, I have a question about the input of  LSTM layer.
The Keras Document says that the input data should be 3D tensor with shape (nb_samples, timesteps, input_dim). In my LSTM model, I set the timesteps as 50, but the model doesn't perform well. Is there any range limitations of 'timesteps'? Like smaller than a number ? 
Can LSTM layer can use all the time information of each sequence?
Thank you in advance!

**Comment by viksit** [comment]
What kind of data do you have, and could you shed some more light on what isn't working well?

AFAIK, there's no "limitation" on the timestep number.

**Comment by fchollet** [comment]
No limitations, but for the sake of performance it's better to keep it
fairly small (aim at <200).

**Comment by Yingyingzhang15** [comment]
@viksit  Thank you for your answer! My training data size is 45000*50*3, where 45000 is the nb_sample, 50 is timesteps and 3 is the input_dim. There are 3 classes to classify and I want to give every 50*3 vector a label.  My validation data size is 7000*50*3.

Since  I'm new to Keras, I just tried the two layer model and get the training accuracy 96%, validation accuracy 80%.   I hope to get the accuracy of 95%. So should I just increase the number of layers?

There are many parameters(batch_size, epoch,dropout...) , it takes a lot of time to try different parameters. Could  you give me some good advice, please?


My models:

model = Sequential()
model.add(LSTM(150, input_shape=(tsteps, 3), dropout_W=0.5, dropout_U=0.1))
model.add(Dense(3))
model.add(Activation('softmax'))
model.compile(loss='binary_crossentropy', optimizer='adam')

Thank you!

**Comment by Yingyingzhang15** [comment]
@fchollet  Thank you !
